File: user-service/src/main/java/com/gallery/userservice/infrastructure/UserEntity.java
package com.gallery.userservice.infrastructure;

import com.gallery.userservice.domain.User;
import jakarta.persistence.*;

@Entity
@Table(name = "USERS")
public class UserEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String role;

    private String phone;

    public UserEntity() {}

    public UserEntity(User user) {
        this.id = user.getId().getId();
        this.name = user.getName();
        this.role = user.getRole();
        this.phone = user.getPhone();
    }

    public User toDomain() {
        return new User(
                new com.gallery.userservice.domain.UserID(this.id),
                this.name,
                this.role,
                this.phone
        );
    }

    // Getters & Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getRole() { return role; }
    public void setRole(String role) { this.role = role; }

    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; }
}

File: app/UserService/Infrastructure/Entities/UserEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using UserService.Domain;

namespace UserService.Infrastructure.Entities
{
    [Table("USER", Schema = "dbo")]
    public class UserEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("Id")]
        public int Id { get; set; }

        [Column("Name")]
        [Required]
        [StringLength(100)]
        public string Name { get; set; }

        [Column("Role")]
        [Required]
        [StringLength(50)]
        public string Role { get; set; }

        [Column("Phone")]
        [StringLength(20)]
        public string? Phone { get; set; }

        [Column("Email")]
        [Required]
        [StringLength(100)]
        public string Email { get; set; }

        [Column("PasswordHash")]
        [Required]
        [StringLength(255)]
        public string PasswordHash { get; set; }

        public UserEntity()
        {
            Id = 0;
            Name = "";
            Role = "";
            Phone = "";
            Email = "";
            PasswordHash = "";
        }

        public UserEntity(User user)
        {
            Id = user.Id;
            Name = user.Name;
            Role = user.Role;
            Phone = user.Phone;
            Email = user.Email;
            PasswordHash = user.PasswordHash;
        }

        public User ToUser()
        {
            return new User(Id, Name, Role, Phone, Email, PasswordHash);
        }
    }
}